src/utils.js

```diff
@@ -183,4 +183,25 @@ global.utils = {
     }
     return arrayParts;
   },
+
+  /**
+   * Given a direction, 1-8, increment it by some value and wrap back to 1-8
+   *
+   * @param {Number} direction
+   * @param {Number} change
+   * @return {Number}
+   */
+  changeDirection: function(direction, change) {
+    return ((direction + change - 1) % 8) + 1;
+  },
+  /**
+   * Given a direction, 1-8, return the opposite direction
+   *
+   * @param {Number} direction
+   * @return {Number}
+   */
+  oppositeDirection: function(direction) {
+    return ((direction + 3) % 8) + 1;
```

Comment:
equal to ``global.utils.changeDirection(direction, 4);`` ?

Reply:
yes, but "4" is a not-entirely-scrutable magic number there; doesn't convey the same concept as "opposite". also, tiny optimization to remove as little arithmetic, since we use oppositeDirection far more often than changeDirection.

Reply:
💯